crates/node/src/networking/p2p.rs

```diff
@@ -175,9 +313,141 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_peer_list_inter_connection() {
+        //start_logger(LevelFilter::ERROR);
+
+        let (tx1, mut rx1) = mpsc::channel(1);
+        let (tx2, mut rx2) = mpsc::channel(1);
+        let (tx3, mut rx3) = mpsc::channel(1);
+        let (sink1, sink2, sink3) = (
+            Arc::new(Sink::new(Arc::new(tx1))),
+            Arc::new(Sink::new(Arc::new(tx2))),
+            Arc::new(Sink::new(Arc::new(tx3))),
+        );
+        let (peer1, peer2, peer3) = (
+            P2P::new("peer1", "127.0.0.1:0".parse().unwrap(), "secret passphrase").await,
+            P2P::new("peer2", "127.0.0.1:0".parse().unwrap(), "secret passphrase").await,
+            P2P::new("peer3", "127.0.0.1:0".parse().unwrap(), "secret passphrase").await,
+        );
+
+        tracing::debug!("start listening");
+        let bind_add = peer1.node().start_listening().await.expect("peer1 listen");
+        let bind_add = peer2.node().start_listening().await.expect("peer2 listen");
+        let bind_add = peer3.node().start_listening().await.expect("peer3 listen");
+
+        tracing::debug!("register tx handlers");
+        peer1.register_tx_handler(sink1.clone()).await;
+        peer2.register_tx_handler(sink2.clone()).await;
+        peer3.register_tx_handler(sink3.clone()).await;
+
+        tracing::debug!("connect peer2 to peer1");
+        peer2
+            .node()
+            .connect(peer1.node().listening_addr().unwrap())
+            .await
+            .unwrap();
+
+        tracing::debug!("connect peer3 to peer1");
+        peer3
+            .node()
+            .connect(peer1.node().listening_addr().unwrap())
+            .await
+            .unwrap();
+
+        tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
+
+        //assert_eq!(peer1.get_connected_peers().len(), 3);
+        //assert_eq!(peer2.get_connected_peers().len(), 3);
+        //assert_eq!(peer3.get_connected_peers().len(), 3);
```

Comment:
Should these be uncommented? If not, maybe remove them?

Reply:
Yes, I'll remove them.